src/hooks/useTransactionList.ts

```diff
@@ -25,10 +25,17 @@ interface TransactionListOptions {
  * until `atEnd` becomes true.
  */
 export function useTransactionList(wallet: EdgeCurrencyWallet, tokenId: EdgeTokenId, opts: TransactionListOptions = {}): Output {
-  const { searchString = '', spamThreshold = '0' } = opts
+  const { searchString = '', spamThreshold: spamThresholdProp = '0' } = opts
 
   const requestMore = React.useRef(() => {})
 
+  // Ignore changes to the spamThreshold unless it's changing to or from 0
+  // This prevents starting the stream over when the exchange rate changes
+  const [spamThreshold, setSpamThreshold] = React.useState<string>(spamThresholdProp)
+  if (spamThresholdProp !== spamThreshold && (spamThresholdProp === '0' || spamThreshold === '0')) {
+    setSpamThreshold(spamThresholdProp)
+  }
```

Comment:
You can't update state variables during the render. It makes React complain.

Instead, `useRef` can do the same thing:

```ts
const spamThreshold = React.useRef(spamThresholdProp)
if (spamThresholdProp !== spamThreshold && (spamThresholdProp === '0' || spamThreshold === '0')) {
    spamThreshold.current = spamThreshold
}
```

Actually, since we are turning `spamThreshold` into a ref, we need to add `.current` everywhere we use it. For this reason, I propose renaming `spamThresholdProp` back to `spamThreshold`, and then naming the ref `spamThresholdRef`, so it's obvious we need the `.current` when we access it.

Reply:
This is how I did it the first time and then switched it to useState. My concern at the time was not updating if the spam option was toggled but that's why the 0 checks are there so I apparently forgot I addressed that case.